src/utils/core.js

```diff
@@ -63,6 +63,15 @@ export function isIntegralNumber(x) {
     return Number.isInteger(x) || typeof x === 'bigint'
 }
 
+/**
+ * Determine if a provided width or height is nullish.
+ * @param {*} x The value to check.
+ * @returns {boolean} True if the value is `null`, `undefined` or `-1`, false otherwise.
+ */
+export function isNullishDimension(x) {
+    return x === null || x === undefined || x === -1 || x === '-1';
```

Comment:
Being verbose works for me! Any reason to include the string `'-1'`?

Reply:
No reason other than I don't trust users, and JavaScript can be awkward.  
For instance, someone might have an implementation that uses `prompt` or something weird that returns a string so I figured it would be easier to handle it.

Happy to remove it, however.